react_juce/core/ShadowView.h

```diff
@@ -160,147 +106,56 @@ namespace reactjuce
-                }
-
-                if (props[layoutAnimatedProp].isObject())
-                {
-                    double const durationMs = props[layoutAnimatedProp].getProperty(durationProp, 50.0);
-                    double const frameRate = props[layoutAnimatedProp].getProperty(frameRateProp, 45);
-                    int const et = props[layoutAnimatedProp].getProperty(easingProp, 0);
-
-                    return flushViewLayoutAnimated(durationMs, static_cast<int> (frameRate), static_cast<BoundsAnimator::EasingType>(et));
-                }
-            }
-
-            view->setFloatBounds(getCachedLayoutBounds());
-            view->setBounds(getCachedLayoutBounds().toNearestInt());
-
-            for (auto& child : children)
-                child->flushViewLayout();
-        }
+        virtual void flushViewLayout();
 
         /** Recursive traversal of the shadow tree, flushing layout bounds to the
          *  associated view components smoothly over time. This recursive step allows
          *  an animation of a component subtree by just marking the parent as animated.
          */
-        virtual void flushViewLayoutAnimated(double const durationMs, int const frameRate, BoundsAnimator::EasingType const et)
-        {
-            auto viewCurrentBounds = view->getBounds().toFloat();
-            auto viewDestinationBounds = getCachedLayoutBounds();
-
-            animator = std::make_unique<BoundsAnimator>(
-                durationMs,
-                frameRate,
-                et,
-                viewCurrentBounds,
-                viewDestinationBounds,
-                [safeView = juce::Component::SafePointer(view)](auto && stepBounds) {
-                    if (auto* v = safeView.getComponent()) {
-                        v->setFloatBounds(stepBounds);
-                        v->setBounds(stepBounds.toNearestInt());
-                    }
-                }
-            );
-
-            for (auto& child : children) {
-                child->flushViewLayoutAnimated(durationMs, frameRate, et);
-            }
-        }
+        virtual void flushViewLayoutAnimated(double durationMs, int frameRate, BoundsAnimator::EasingType et);
 
     protected:
         //==============================================================================
-        YGNodeRef yogaNode;
-        View* view = nullptr;
-        juce::NamedValueSet props;
+        std::vector<ShadowView*>& getChildren();
 
-        std::unique_ptr<BoundsAnimator> animator;
-        std::vector<ShadowView*> children;
+        //==============================================================================
 
     private:
+        class Pimpl;
+        std::unique_ptr<Pimpl> mPimpl;
+
+        //TODO: Damn. Is there a better way to do what is required here? Is this the end of the world?
+        //            Probably not in this very particular case.
+        friend Pimpl;
```

Comment:
@nick-thompson , This isn't wonderful but I think it's OK as completely hidden and doesn't really break encapsulation. It means that the ShadowView yoga pimpl can use the likes of `childView->mPimpl->yogaNode` in `ShadowView_Yoga.cpp`